### DISCUSSION TITLE
Fonts Error

### DISCUSSION THREAD

**Discussion by Xiuyuan121** [JanDeDobbeleer/oh-my-posh#3949] [Troubleshoot]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When I use Oh My Posh in Windows Terminal, the characters appear garbled, how can I solve this?

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/78246617/f825e922-df09-4c6a-bf40-6d74d24ef384)

### Theme

default

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 16.9.1

Shell: powershell (5.1.19041.3031)

Segments:

ConsoleTitle(true)                        -   0 ms - powershell in xiuyu
session(true)                             -   0 ms -  xiuyu
path(true)                                -   0 ms -   ~
git(false)                                -  17 ms -
root(false)                               -   0 ms -
exit(true)                                -   2 ms -  
node(false)                               -   0 ms -
go(false)                                 -   0 ms -
python(false)                             -   0 ms -
shell(true)                               -   0 ms - in powershell
time(true)                                -   0 ms - at 12:22:26

Run duration: 26.2761ms

Cache path: C:\Users\xiuyu\AppData\Local\oh-my-posh

Config path: no --config set, using default built-in configuration

**Comment by JanDeDobbeleer** [comment]
hey @XinghangGuan, are you certain the font name is correct? A screenshot of the garbled prompt would be useful. Usually it's two things:

- incorrect font name
- encoding issue (Windows settings)

**Comment by Xiuyuan121** [comment]
Yes, it was true

**Comment by Xiuyuan121** [comment]
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/78246617/a2b332d7-e30e-4a06-9f77-f58de67af04d)

**Comment by Xiuyuan121** [comment]
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/78246617/0a860291-e253-492c-9586-c556a6e685e8)

**Comment by Xiuyuan121** [comment]
hey @JanDeDobbeleer 
so
what can i do?
i'm in a hurry

**Reply by JanDeDobbeleer** [reply]
You need to validate the font name and how it's installed. In my case, user installed fonts to not work. And font names can change between Nerd Font major releases, so double check. But it's a configuration issue I can't fix without access to your computer, this is something you need to figure out (or ask the Terminal team).

**Comment by Xiuyuan121** [comment]
OH It's RUN!!!
But why it will show this?
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/78246617/3b4737f6-3fff-4184-bb81-9dad5c308b9b)

@JanDeDobbeleer

**Reply by JanDeDobbeleer** [reply]
@XinghangGuan that's the [transient prompt](https://ohmyposh.dev/docs/configuration/transient) feature in the default theme.

**Reply by Xiuyuan121** [reply]
Can we cancel it then?
And after I press Enter, it waits nearly a second for a new line to open
@JanDeDobbeleer

**Reply by JanDeDobbeleer** [reply]
@XinghangGuan you can [export and edit the configuration](https://ohmyposh.dev/docs/installation/customize#adjust-a-theme), yes.

**Reply by Xiuyuan121** [reply]
but why after I press Enter, it waits nearly a second for a new line to open

**Reply by JanDeDobbeleer** [reply]
@XinghangGuan that's probably a virus scanner blocking oh-my-posh' execution

**Reply by Xiuyuan121** [reply]
@JanDeDobbeleer so what can i do?

**Reply by JanDeDobbeleer** [reply]
@XinghangGuan validate what's running on your machine? This IT support is getting a bit beyond the scope of the project.

**Comment by Xiuyuan121** [comment]
ok,thank you!

**Comment by Xiuyuan121** [comment]
I've found that no matter which theme I use, there can be problems
For example, I used this command:
$Theme = Get-ChildItem $env: UserProfile\\ AppData\\ Local\\ Programs\\ oh-my-posh\\ themes\\ | Get-Random
Echo "hello! today's lucky theme is: $theme:) "
oh-my-posh --init --shell pwsh --config $theme. FullName | Invoke-Expression
But when I started it, I still found that it had the following problems:
1. The icon zoom is incorrect, and some can only display the upper left corner.
2. After I press the enter key, the next line will still pop up after about 1 second, and the previous line will disappear at the same time
........
I don't know the rest
So
What can I do?

**Reply by JanDeDobbeleer** [reply]
@XinghangGuan icon displaying is a Terminal task, so you'll have to reach out there. About slowness, 2 things to validate:
- segment being slow: in the folder wher eit's slow, run `oh-my-posh debug`. That will show exactly what time is spent where, also aggregated per segment. If that one runs < 100ms, it's not this but:
- oh-my-posh execution blocked by a virus scanner/defender: even while being code signed with a valid (and seriously expensive) certificate, it can still happen. Exclude the binary path from the virus scanner/defender's scan list.

**Comment by Xiuyuan121** [comment]
@JanDeDobbeleer

**Comment by Xiuyuan121** [comment]
Hey,friend.
Long time no see.
I finally have time to solve this problem.
I run _oh-my-posh debug_ ,it is displayed as follows:
ConsoleTitle(true)  -   0 ms
Session(true)       -   0 ms
Path(true)          -   0 ms
Git(false)          -  24 ms
Node(false)         -   0 ms
Go(false)           -   0 ms
Julia(false)        -   0 ms
Python(false)       -   0 ms
Ruby(false)         -   0 ms
Azfunc(false)       -   0 ms
Aws(false)          -   0 ms
Root(false)         -   0 ms
Executiontime(true) -   0 ms
Status(true)        -   1 ms
Shell(true)         -   0 ms
Ytm(false)          -   2 ms
Battery(true)       -  68 ms
Time(true)          -   0 ms
why Git(false) and Battery(true) used so much time?
p.s: I live in China,so i can't use Git simply and I added it to the antivirus trust zone.

**Reply by JanDeDobbeleer** [reply]
@Xiuyuan121 to debug that, we'll need to full output of `oh-my-posh debug`.

**Comment by Xiuyuan121** [comment]
OK,here is:
Segments:

ConsoleTitle(true) -   2 ms
Session(true)      -   0 ms
Path(true)         -   1 ms
Git(false)         -  24 ms
Root(false)        -   0 ms
Status(true)       -   1 ms
Node(false)        -   0 ms
Go(false)          -   0 ms
Python(false)      -   0 ms
Shell(true)        -   0 ms
Time(true)         -   0 ms

Run duration: 35.768ms

Cache path: C:\Users\xiuyu\AppData\Local\oh-my-posh

Config path: no --config set, using default built-in configuration

Logs:

[DEBUG] 18:21:49.524 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.524 shell.go:Getenv(TERM_PROGRAM) - 0s
[DEBUG] 18:21:49.524 shell.go:Getenv:399 → C:\Users\xiuyu\AppData\Local
[TRACE] 18:21:49.524 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 18:21:49.525 shell_windows.go:CachePath() - 533.6µs
[DEBUG] 18:21:49.526 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.526 shell.go:Getenv(POSH_THEME) - 108.3µs
[DEBUG] 18:21:49.526 shell.go:resolveConfigPath:319 → No config set, fallback to default config
[TRACE] 18:21:49.526 shell.go:resolveConfigPath() - 108.3µs
[TRACE] 18:21:49.526 shell.go:Init() - 1.6773ms
[TRACE] 18:21:49.526 shell.go:Flags() - 0s
[DEBUG] 18:21:49.526 config.go:loadConfig:129 → no config file specified, using default
[TRACE] 18:21:49.526 config.go:loadConfig() - 0s
[TRACE] 18:21:49.526 shell.go:Flags() - 0s
[DEBUG] 18:21:49.526 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.526 shell.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 18:21:49.526 shell_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[TRACE] 18:21:49.526 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.526 shell_windows.go:WindowsRegistryKeyValue:215 → ColorizationColor(DWORD): 0xC40078D7
[DEBUG] 18:21:49.526 shell.go:Shell:662 → no shell name provided in flags, trying to detect it
[DEBUG] 18:21:49.531 shell.go:Shell:670 → process name: powershell.exe
[TRACE] 18:21:49.531 shell.go:Shell() - 4.8464ms
[DEBUG] 18:21:49.531 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.531 shell.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 18:21:49.531 debug.go:PrintDebug:22 → Segment: Title
[DEBUG] 18:21:49.531 text.go:Render:70 → Rendering template: {{ .Shell }} in {{ .Folder }}
[TRACE] 18:21:49.531 shell_windows.go:Root() - 0s
[TRACE] 18:21:49.531 shell.go:Shell() - 0s
[TRACE] 18:21:49.531 shell.go:StatusCodes() - 0s
[TRACE] 18:21:49.531 shell_windows.go:IsWsl() - 0s
[DEBUG] 18:21:49.533 shell.go:TemplateCache:843 → environment: [ALLUSERSPROFILE=C:\ProgramData APPDATA=C:\Users\xiuyu\AppData\Roaming asl.log=Destination=file CommonProgramFiles=C:\Program Files\Common Files CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files CommonProgramW6432=C:\Program Files\Common Files COMPUTERNAME=LAPTOP-2PMJSS3 ComSpec=C:\WINDOWS\system32\cmd.exe DriverData=C:\Windows\System32\Drivers\DriverData HOMEDRIVE=C: HOMEPATH=\Users\xiuyu IGCCSVC_DB=AQAAANCMnd8BFdERjHoAwE/Cl+sBAAAAQMNgSuz4aUWKXW4TAv64DQQAAAACAAAAAAAQZgAAAAEAACAAAADX+JCm7X3lyrfN5W/HXWeADS1rHrLkcMcQitNkxeqK/gAAAAAOgAAAAAIAACAAAAAU05+CwkYHjDYOvVXZk12bqK0RszBmccLU24bKNIjivmAAAADjeBGxbaWsKkMTeUnqZyX3IRF8FP2xzpaUeCksG4WbkpY/0Mw3I0kvSUbXh+6khbOSLvpxhuB663MW9eMDsGckWoddDouyydvZaFX4MH0PdcT4E4wy6UZwwqJewxOmRdNAAAAAndrWpKzzXa/suvmDKWM0eFGdM06OC8NNaQUJ8UtRerHxvPe+7NSzzbhgyz1stSWsNsDQH+RORrldQDh52F0jTw== INTEL_DEV_REDIST=C:\Program Files (x86)\Common Files\Intel\Shared Libraries\ LOCALAPPDATA=C:\Users\xiuyu\AppData\Local LOGONSERVER=\\LAPTOP-2PMJSS3 MIC_LD_LIBRARY_PATH=C:\Program Files (x86)\Common Files\Intel\Shared Libraries\compiler\lib\mic NUMBER_OF_PROCESSORS=16 OneDrive=C:\Users\xiuyu\OneDrive OS=Windows_NT Path=C:\Program Files (x86)\VMware\VMware Workstation\bin\;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;D:\Program Files\Python311\Scripts\;D:\Program Files\Python311\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;G:\MinGW\bin;C:\Program Files\dotnet\;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Bandizip\;C:\Program Files\Calibre2\;F:\CMake\bin;E:\VideoDownload;C:\Users\xiuyu\AppData\Local\Microsoft\WindowsApps;;%CLion%;G:\Microsoft VS Code\bin;C:\Users\xiuyu\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\xiuyu\AppData\Local\Programs\oh-my-posh\bin;D:\oh-my-posh\bin PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.CPL POSH_INSTALLER=manual POSH_THEMES_PATH=D:\oh-my-posh\themes PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 141 Stepping 1, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=8d01 ProgramData=C:\ProgramData ProgramFiles=C:\Program Files ProgramFiles(x86)=C:\Program Files (x86) ProgramW6432=C:\Program Files PSModulePath=C:\Users\xiuyu\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules PUBLIC=C:\Users\Public SESSIONNAME=Console SystemDrive=C: SystemRoot=C:\WINDOWS TEMP=C:\Users\xiuyu\AppData\Local\Temp TMP=C:\Users\xiuyu\AppData\Local\Temp USERDOMAIN=LAPTOP-2PMJSS3 USERDOMAIN_ROAMINGPROFILE=LAPTOP-2PMJSS3 USERNAME=xiuyu USERPROFILE=C:\Users\xiuyu windir=C:\WINDOWS WSLENV=WT_SESSION:WT_PROFILE_ID: WT_PROFILE_ID={61c54bbd-c2c6-5271-96e7-009a87ff44bf} WT_SESSION=800e83c7-40a4-4d5f-b0a9-5d9d9fb1d97b ZES_ENABLE_SYSMAN=1]
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.533 shell.go:Pwd:429 → C:\Users\xiuyu
[TRACE] 18:21:49.533 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.533 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.533 shell.go:User:564 → xiuyu
[TRACE] 18:21:49.533 shell.go:User() - 0s
[DEBUG] 18:21:49.533 shell.go:Host:576 → LAPTOP-2PMJSS3
[TRACE] 18:21:49.533 shell.go:Host() - 0s
[TRACE] 18:21:49.533 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.533 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.533 shell.go:Getenv(SHLVL) - 0s
[TRACE] 18:21:49.533 shell.go:TemplateCache() - 1.658ms
[DEBUG] 18:21:49.533 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.533 shell.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 18:21:49.533 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.533 shell.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 18:21:49.533 shell.go:Flags() - 0s
[TRACE] 18:21:49.533 shell.go:Flags() - 0s
[TRACE] 18:21:49.533 shell.go:Shell() - 0s
[TRACE] 18:21:49.533 shell.go:Shell() - 0s
[TRACE] 18:21:49.533 shell.go:Flags() - 0s
[TRACE] 18:21:49.533 shell.go:Flags() - 0s
[TRACE] 18:21:49.533 shell.go:Flags() - 0s
[TRACE] 18:21:49.533 shell.go:Flags() - 0s
[TRACE] 18:21:49.534 shell.go:Pwd() - 0s
[TRACE] 18:21:49.534 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.534 properties.go:GetString:28 → {{ .Code }}
[TRACE] 18:21:49.534 shell.go:Pwd() - 0s
[TRACE] 18:21:49.534 shell.go:Pwd() - 0s
[TRACE] 18:21:49.534 shell.go:Flags() - 0s
[DEBUG] 18:21:49.534 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.534 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.534 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.534 shell_windows.go:56 → C:\Users\xiuyu
[DEBUG] 18:21:49.534 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.534 segment.go:SetEnabled:518 → Segment: Root
[DEBUG] 18:21:49.534 segment.go:SetEnabled:518 → Segment: Git
[DEBUG] 18:21:49.534 segment.go:SetEnabled:518 → Segment: Session
[DEBUG] 18:21:49.534 segment.go:SetEnabled:518 → Segment: Status
[TRACE] 18:21:49.534 shell.go:StatusCodes() - 0s
[DEBUG] 18:21:49.534 shell.go:Getenv:399 → NO DATA
[TRACE] 18:21:49.534 shell.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 18:21:49.534 shell.go:Getenv:399 → NO DATA
[DEBUG] 18:21:49.534 text.go:Render:70 → Rendering template: {{ .Code }}
[TRACE] 18:21:49.534 shell.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.534 shell_windows.go:Root() - 0s
[DEBUG] 18:21:49.534 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.534 segment.go:SetEnabled:518 → Segment: Path
[TRACE] 18:21:49.534 shell.go:Pwd() - 0s
[TRACE] 18:21:49.534 shell.go:Shell() - 0s
[TRACE] 18:21:49.534 shell.go:Shell() - 0s
[TRACE] 18:21:49.534 shell.go:Flags() - 0s
[DEBUG] 18:21:49.534 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 18:21:49.534 properties.go:GetString:28 → 
[DEBUG] 18:21:49.534 properties.go:GetString:28 → 
[DEBUG] 18:21:49.534 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.534 properties.go:GetString:28 → ~
[DEBUG] 18:21:49.534 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 18:21:49.534 shell.go:PathSeparator() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:TemplateCache() - 0s
[DEBUG] 18:21:49.534 properties.go:GetBool:22 → always_enabled: true
[TRACE] 18:21:49.534 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:PathSeparator() - 0s
[TRACE] 18:21:49.534 shell.go:PathSeparator() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:PathSeparator() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:PathSeparator() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:GOOS() - 0s
[TRACE] 18:21:49.534 shell.go:PathSeparator() - 0s
[TRACE] 18:21:49.534 shell.go:Pwd() - 0s
[TRACE] 18:21:49.534 shell_windows.go:IsWsl() - 0s
[TRACE] 18:21:49.534 shell.go:StackCount() - 0s
[DEBUG] 18:21:49.535 win32_windows.go:isWriteable:299 → not current user or in group
[DEBUG] 18:21:49.535 win32_windows.go:isWriteable:303 → current user is member of S-1-5-32-544
[DEBUG] 18:21:49.535 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 18:21:49.535 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 18:21:49.535 shell_windows.go:DirIsWritable() - 542.8µs
[TRACE] 18:21:49.535 shell.go:TemplateCache() - 0s
[ERROR] 18:21:49.558 shell.go:CommandPath:618 → exec: "git.exe": executable file not found in %PATH%
[TRACE] 18:21:49.558 shell.go:CommandPath(git.exe) - 24.166ms
[TRACE] 18:21:49.558 shell.go:HasCommand(git.exe) - 24.166ms
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template:  {{ if gt .Code 0 }}{{ else }}{{ end }}
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template:   {{ path .Path .Location }}
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template:  {{ if .SSHSession }} {{ end }}{{ .UserName }}
[TRACE] 18:21:49.558 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.558 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.558 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.558 shell.go:Shell() - 0s
[TRACE] 18:21:49.558 shell.go:Shell() - 0s
[TRACE] 18:21:49.558 shell.go:Shell() - 0s
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template: diamond
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template: powerline
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template: powerline
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template: powerline
[DEBUG] 18:21:49.558 text.go:Render:70 → Rendering template: {{ if gt .Code 0 }}p:red{{ end }}
[TRACE] 18:21:49.559 shell.go:TemplateCache() - 0s
[DEBUG] 18:21:49.559 text.go:Render:70 → Rendering template: diamond
[TRACE] 18:21:49.559 shell.go:Shell() - 0s
[TRACE] 18:21:49.559 shell.go:Shell() - 0s
[TRACE] 18:21:49.559 shell.go:Shell() - 0s
[TRACE] 18:21:49.559 shell.go:Shell() - 0s
[TRACE] 18:21:49.559 shell.go:Flags() - 0s
[TRACE] 18:21:49.559 shell.go:Flags() - 0s
[TRACE] 18:21:49.559 shell.go:Flags() - 0s
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[TRACE] 18:21:49.559 shell.go:GOOS() - 0s
[TRACE] 18:21:49.559 shell.go:Flags() - 0s
[DEBUG] 18:21:49.559 segment.go:SetEnabled:518 → Segment: Time
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[TRACE] 18:21:49.559 shell.go:Flags() - 0s
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.559 shell.go:GOOS() - 0s
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.559 segment.go:SetEnabled:518 → Segment: Node
[TRACE] 18:21:49.559 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.559 segment.go:SetEnabled:518 → Segment: Go
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.559 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.559 segment.go:SetEnabled:518 → Segment: Shell
[DEBUG] 18:21:49.559 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[]
[TRACE] 18:21:49.559 shell.go:Shell() - 0s
[TRACE] 18:21:49.559 shell.go:Flags() - 0s
[TRACE] 18:21:49.559 shell.go:TemplateCache() - 0s
[DEBUG] 18:21:49.559 properties.go:GetString:28 → 15:04:05
[TRACE] 18:21:49.559 shell.go:TemplateCache() - 0s
[DEBUG] 18:21:49.559 properties.go:GetStringArray:52 → extensions: [*.go go.mod]
[DEBUG] 18:21:49.559 properties.go:GetStringArray:52 → folders: []
[DEBUG] 18:21:49.559 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 18:21:49.559 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .npmrc .vue]
[DEBUG] 18:21:49.559 properties.go:GetStringArray:52 → folders: []
[DEBUG] 18:21:49.559 properties.go:GetBool:22 → home_enabled: false
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[DEBUG] 18:21:49.559 properties.go:GetString:28 → files
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[TRACE] 18:21:49.559 shell.go:GOOS() - 0s
[DEBUG] 18:21:49.559 segment.go:SetEnabled:518 → Segment: Python
[DEBUG] 18:21:49.559 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 18:21:49.559 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 18:21:49.559 properties.go:GetBool:22 → home_enabled: false
[TRACE] 18:21:49.559 shell.go:Pwd() - 0s
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[DEBUG] 18:21:49.559 shell_windows.go:56 → C:\Users\xiuyu
[DEBUG] 18:21:49.559 text.go:Render:70 → Rendering template: at <p:blue><b>{{ .CurrentDate | date "15:04:05" }}</b></>
[DEBUG] 18:21:49.559 text.go:Render:70 → Rendering template: in <p:blue><b>{{ .Name }}</b></>
[TRACE] 18:21:49.559 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.560 shell.go:Shell() - 0s
[TRACE] 18:21:49.560 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.560 shell.go:Shell() - 0s
[DEBUG] 18:21:49.560 text.go:Render:70 → Rendering template: plain
[DEBUG] 18:21:49.560 text.go:Render:70 → Rendering template: plain
[DEBUG] 18:21:49.560 text.go:Render:70 → Rendering template: plain
[DEBUG] 18:21:49.560 text.go:Render:70 → Rendering template: plain
[DEBUG] 18:21:49.560 text.go:Render:70 → Rendering template: plain
[TRACE] 18:21:49.560 shell.go:Shell() - 0s
[TRACE] 18:21:49.560 shell.go:Shell() - 0s
[TRACE] 18:21:49.560 shell.go:Flags() - 0s
[DEBUG] 18:21:49.560 text.go:Render:70 → Rendering template: {{ .Shell }} in {{ .Folder }}
[TRACE] 18:21:49.560 shell.go:TemplateCache() - 0s
[TRACE] 18:21:49.560 shell.go:Shell() - 0s
[TRACE] 18:21:49.560 shell.go:Shell() - 0s
[DEBUG] 18:21:49.560 shell_windows.go:TerminalWidth:109 → terminal width: 120
[TRACE] 18:21:49.560 shell_windows.go:TerminalWidth() - 0s
[DEBUG] 18:21:49.560 shell.go:Getenv:399 → C:\Users\xiuyu\AppData\Local
[TRACE] 18:21:49.560 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 18:21:49.561 shell_windows.go:CachePath() - 525.6µs
[TRACE] 18:21:49.561 shell.go:Flags() - 0s

**Reply by JanDeDobbeleer** [reply]
The git segment is "slow" because you don't have git installed. 24 milliseconds won't be noticeable, but there's no point in using the git segment when you don't have git installed.

```
[ERROR] 18:21:49.558 shell.go:CommandPath:618 → exec: "git.exe": executable file not found in %PATH%
[TRACE] 18:21:49.558 shell.go:CommandPath(git.exe) - 24.166ms
[TRACE] 18:21:49.558 shell.go:HasCommand(git.exe) - 24.166ms
```

**Comment by Xiuyuan121** [comment]
So i need install a git?

**Reply by JanDeDobbeleer** [reply]
If you want to use the git segment it's required that git is on the system, yes. Otherwise there's no point in using it (assuming you're also working in a git repository).

**Reply by DungRaymond** [reply]
Come on dude what kind of naive question is that are you even a computer guy?

**Comment by Xiuyuan121** [comment]
If I don't use Git, how can I improve the loading speed of oh-my-posh?

**Comment by Xiuyuan121** [comment]
I just installed Git and found that the startup time was reduced to about 500ms, but the line wrapping demo problem I told you about earlier still persists, even switching the theme doesn't help.

**Reply by JanDeDobbeleer** [reply]
Can you show a screenshot of what you mean?

**Reply by DungRaymond** [reply]
Powershell is known for average 900ms start up time with bunches of stuffs installed to start along with it and you have 500ms. Stop whining little guy. If you want 90-200ms start up time use vanilla powershell.

**Comment by Xiuyuan121** [comment]
you can see the chat at 20230613,like it
> but why after I press Enter, it waits nearly a second for a new line to open
and i'm sure,no any apps make it
